artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/ActiveMQServerControlImpl.java

```diff
@@ -4479,5 +4483,45 @@ public void replay(String startScan, String endScan, String address, String targ
 
       server.replay(startScanDate, endScanDate, address, target, filter);
    }
+
+   @Override
+   public void stopEmbeddedWebServer() throws Exception {
+      getEmbeddedWebServerComponent().stop(true);
+   }
+
+   @Override
+   public void startEmbeddedWebServer() throws Exception {
+      getEmbeddedWebServerComponent().start();
+   }
+
+   @Override
+   public void restartEmbeddedWebServer() throws Exception {
+      /*
+       * This needs to be run in its own thread managed by the broker because if it is run on a thread managed by Jetty
+       * (e.g. if it is invoked from the web console) then the thread will die before Jetty can be restarted.
+       */
+      server.getThreadPool().execute(() -> {
+         try {
+            stopEmbeddedWebServer();
+            startEmbeddedWebServer();
```

Comment:
Thinking about this more I still think these need coordinated here in some way. If I call restart, and then I immediately call stop from the same thread, the webserver should end up stopped. As this is now, because they are handled in different threads it seems just as likely it will end up [re]started instead.

Regardless of approach taken to addressing that, it feels like maybe it should even also await the operation being completed for some period before returning. It could even throw on failure like start and stop would.

Reply:
Done.